src/Apps/Order2/Routes/Checkout/Components/Order2PaymentStep/Order2PaymentForm.tsx

```diff
@@ -0,0 +1,243 @@
+  type StripePaymentElementOptions,
+  loadStripe,
+} from "@stripe/stripe-js"
+import { Collapse } from "Apps/Order/Components/Collapse"
+import { FadeInBox } from "Components/FadeInBox"
+import { getENV } from "Utils/getENV"
+import { useState } from "react"
+import LockIcon from "@artsy/icons/LockIcon"
+import { RouterLink } from "System/Components/RouterLink"
+import createLogger from "Utils/logger"
+import { graphql, useRelayEnvironment, fetchQuery } from "react-relay"
+import type { Order2PaymentFormConfirmationTokenQuery } from "__generated__/Order2PaymentFormConfirmationTokenQuery.graphql"
+
+const stripePromise = loadStripe(getENV("STRIPE_PUBLISHABLE_KEY"))
+const logger = createLogger("Order2PaymentForm")
+
+interface Props {
+  order: any
+  setConfirmationToken: (token: any) => void
+}
+
+export const Order2PaymentForm = ({ order, setConfirmationToken }: Props) => {
+  const { itemsTotal, seller } = order
+
+  if (!itemsTotal) {
+    throw new Error("itemsTotal is required")
+  }
+
+  const orderOptions: StripeElementsUpdateOptions = {
+    amount: itemsTotal.minor,
+    currency: itemsTotal.currencyCode.toLowerCase(),
+    onBehalfOf: seller?.merchantAccount?.externalId,
+  }
+
+  const options: StripeElementsOptions = {
+    mode: "payment",
+    appearance: {
+      variables: {
+        accordionItemSpacing: "10px",
+      },
+      rules: {
+        ".AccordionItem": {
+          border: "none",
+          backgroundColor: "#EFEFEF",
+        },
+        ".AccordionItem:focus-visible": {
+          border: "1px solid black",
+        },
+        ".AccordionItem--selected": {
+          border: "1px solid black",
+        },
+      },
+    },
+    ...orderOptions,
+  }
+
+  return (
+    <Elements stripe={stripePromise} options={options}>
+      <PaymentFormContent setConfirmationToken={setConfirmationToken} />
+    </Elements>
+  )
+}
+
+const PaymentFormContent = ({ setConfirmationToken }) => {
+  const stripe = useStripe()
+  const elements = useElements()
+  const environment = useRelayEnvironment()
+
+  if (!(stripe && elements)) {
+    return null
+  }
+
+  // From Stripe docs
+  const [errorMessage, setErrorMessage] = useState()
+
+  // TODO: move this to CheckoutContext
+  const [selectedPaymentMethod, setSelectedPaymentMethod] = useState("new")
+
+  const paymentElementOptions: StripePaymentElementOptions = {
+    layout: {
+      type: "accordion",
+      spacedAccordionItems: true,
+      defaultCollapsed: true,
+      radios: false,
+    },
+  }
+
+  const onChange = event => {
+    const { elementType, collapsed } = event
+    if (elementType === "payment" && !collapsed) {
+      setSelectedPaymentMethod("new")
+    }
+  }
+
+  const onClickSavedPaymentMethods = () => {
+    setSelectedPaymentMethod("saved")
+    elements?.getElement("payment")?.collapse()
+  }
+
+  const onClickWirePaymentMethods = () => {
+    setSelectedPaymentMethod("wire")
+    elements?.getElement("payment")?.collapse()
+  }
```

Comment:
non-blocking - The function that is a functional component will execute any time it is rendered which could be due to changes in its props or context or some internal state. Each time this happens all functions and variables will be re-declared. Usually this isn't a problem but sometimes when passing a function as a prop down to another component it can become one... but I think some like @damassi  would say that it's actually better to _not_ use that until you are sure it's necessary - the performance hit is negligible compared to the risk you get something wrong and a function gets some stale value, etc.